main.tf

```diff
@@ -52,19 +52,32 @@ resource "aws_launch_configuration" "ecs" {
 }
 
 resource "aws_autoscaling_group" "ecs" {
-  name                 = "asg-${aws_launch_configuration.ecs.name}"
-  vpc_zone_identifier  = ["${var.subnet_id}"]
-  launch_configuration = "${aws_launch_configuration.ecs.name}"
-  min_size             = "${var.min_servers}"
-  max_size             = "${var.max_servers}"
-  desired_capacity     = "${var.servers}"
-  termination_policies = ["OldestLaunchConfiguration", "ClosestToNextInstanceHour", "Default"]
-
-  tags = [{
-    key                 = "Name"
-    value               = "${var.name} ${var.tagName}"
-    propagate_at_launch = true
-  }]
+  name_prefix               = "asg-${aws_launch_configuration.ecs.name}-"
+  vpc_zone_identifier       = ["${var.subnet_id}"]
+  launch_configuration      = "${aws_launch_configuration.ecs.name}"
+  min_size                  = "${var.min_servers}"
+  max_size                  = "${var.max_servers}"
+  desired_capacity          = "${var.servers}"
+  wait_for_capacity_timeout = "${var.wait_for_capacity_timeout}"
```

Comment:
Does this add a similar function to `heartbeat_timeout` ?

Reply:
Not really. It's how long TF will wait for a capacity change before timing out. I added it because we had an ASG I wanted to take from 10 to 1 and it timed out.